src/mvt_sql.py

```diff
@@ -0,0 +1,455 @@
+        if t == "regex":
+            return params[0], "%s ~ '%s'" % (column_name, params[1].replace("'", "\\'"))
+        if t == "true":
+            return params[0], "%s = 'yes'" % (column_name)
+        if t == "untrue":
+            return params[0], "%s = 'no'" % (column_name)
+        if t == "set":
+            return params[0], "%s IS NOT NULL" % (column_name)
+        if t == "unset":
+            return params[0], "%s IS NULL" % (column_name)
+
+        if t == "<":
+            return (
+                params[0],
+                """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) &lt; %s ELSE false END) """
+                % (column_name, column_name, params[1]),
+            )
+        if t == "<=":
+            return (
+                params[0],
+                """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) &lt;= %s ELSE false END)"""
+                % (column_name, column_name, params[1]),
+            )
+        if t == ">":
+            return (
+                params[0],
+                """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) > %s ELSE false END) """
+                % (column_name, column_name, params[1]),
+            )
+        if t == ">=":
+            return (
+                params[0],
+                """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) >= %s ELSE false END) """
+                % (column_name, column_name, params[1]),
+            )
+    except KeyError:
+        pass
+
+
+def get_sql_hints(choosers, obj, zoom):
+    needed = set(
+        [
+            "width",
+            "fill-color",
+            "fill-image",
+            "icon-image",
+            "text",
+            "extrude",
+            "background-image",
+            "background-color",
+            "pattern-image",
+            "shield-text",
+        ]
+    )
+
+    hints = []
+    for chooser in choosers:
+        tags = set()
+        qs = []
+        if not needed.isdisjoint(set(chooser.styles[0].keys())):
+            for rule in chooser.ruleChains:
+                if obj:
+                    if (rule.subject != "") and not _test_feature_compatibility(
+                        obj, rule.subject
+                    ):
+                        continue
+                if not rule.test_zoom(zoom):
+                    continue
+                b = set()
+                for condition in rule.conditions:
+                    q = get_sql(condition, obj)
+                    if q:
+                        if q[1]:
+                            tags.add(q[0])
+                            b.add(q[1])
+
+                if b:
+                    qs.append("(" + " AND ".join(b) + ")")
+
+        if qs:
+            hints.append((tags, " OR ".join(qs)))
+
+    return hints
+
+
+def get_vectors(minzoom, maxzoom, x, y, style, vec):
+    geomcolumn = "way"
+
+    pxtolerance = 0.5
```

Comment:
you have two mechanisms to compensate for zoom/pixel mismatch: you've set px to 0.5 and the tile width to 512 instead of 256. I'd say that tile size should stay 256 and instead here you should use 256/extent or something like that.